### DISCUSSION TITLE
What is the best locatator strategy for Android and iOS testing

### DISCUSSION THREAD

**Discussion by marianafiori** [appium/appium#17774] [Q&A]
For example, I work with an app that is a hybrid, Android and iOS. I'm doing the tests and I wanted to know the best way to get the locators. Can you use the same for both, for example, ID and ACCESSIBILITY_ID? If not, what better strategy to know which locator to take during the test?

**Comment by Higgens75** [comment]
What do you mean with hybrid app ? An app which is programmed with a framework like flutter where one code is written and an app is generated for each plattform ? I think its necessary to know which Multi-Platform Framework ist uses.

**Reply by marianafiori** [reply]
The app is in React, it works for iOS and Android

**Comment by jlipps** [comment]
If it's possible to assign accessibility labels to elements in a cross platform manner, then yes, finding an element by accessibility ID is going to be easiest and fastest.

But if you're using React (not React Native), then you're not dealing with a native app at all and so you will not be able to directly set accessibility labels. You'll be dependent on how the iOS and Android accessibility systems promote information from the HTML source, which could vary and be inconsistent.

**Reply by marianafiori** [reply]
I understood. But what would be the best design pattern strategy for the project? Create different locator files for iOS and Android? I'm working with Python + Pytest.

**Reply by jlipps** [reply]
The strategy that I have used is to have one base object that has locators that work for one platform, and then if necessary a platform specific one which extends it with any differences in locators
